### PR TITLE
make restclient config configurable

### PR DESCRIPTION
**Pull request by shiyan2016**
**What this PR does / why we need it**:
make restclient config configurable

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by shiyan2016** [comment]
@hectorj2f Please help me to retrigger the test,thanks.

File: cmd/controller-manager/app/controller-manager.go [hunk 1]
```diff
@@ -98,6 +99,8 @@ member clusters and do the necessary reconciliation`,
 	flags.StringVar(&kubeFedConfig, "kubefed-config", "", "Path to a KubeFedConfig yaml file. Test only.")
 	flags.StringVar(&kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
 	flags.StringVar(&masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+	flags.IntVar(&restConfigQPS, "rest-config-qps", 5, "QPS of rest config.")
```
Comment (hectorj2f) [inline cmd/controller-manager/app/controller-manager.go:102]: Just to confirm, are these the default values ?

**Comment by hectorj2f** [comment]
I re-trigger it but we are always failing the docker image pulling issue. So we need to fix that first :/.

File: cmd/controller-manager/app/controller-manager.go (hunk 1 above)
Comment (shiyan2016) [reply cmd/controller-manager/app/controller-manager.go:102]: yes, they are default values.

**Comment by hectorj2f** [comment]
This problem is related to this issue https://github.com/kubernetes-sigs/kubefed/issues/1359. We need to work on that.

**Comment by shiyan2016** [comment]
> This problem is related to this issue #1359. We need to work on that.

ok, thanks.

**Comment by CharlesQQ** [comment]
> /retest

1

File: cmd/controller-manager/app/controller-manager.go (hunk 1 above)
Comment (CharlesQQ) [reply cmd/controller-manager/app/controller-manager.go:102]: 1

**Comment by hectorj2f** [comment]
Please, rebase to get the latest changes.

**Comment by shiyan2016** [comment]
> Please, rebase to get the latest changes.

Done it. Thanks.

File: cmd/controller-manager/app/controller-manager.go (hunk 1 above)
Comment (xunpan) [inline cmd/controller-manager/app/controller-manager.go:102]: - if QPS is a float, it is better to use `Float32Var` instead.
- I suggest to use more detailed message for the description. E.g.:
  `QPS indicates the maximum QPS to the api-server from this client`
  `Maximum burst for throttle`

**Review by xunpan — COMMENTED** [review]
please squash 2 commits into 1.

File: cmd/controller-manager/app/controller-manager.go (hunk 1 above)
Comment (shiyan2016) [reply cmd/controller-manager/app/controller-manager.go]: @xunpan Done. Please review it again. Thanks.

**Comment by xunpan** [comment]
@hectorj2f 
I don't know how to approve testing for this PR. Please help on this.

The code is fine to me.

**Comment by irfanurrehman** [comment]
Have trigerred a build.

**Comment by shiyan2016** [comment]
```bash
[Fail] Federated "configmaps" [It] should be created, read, updated and deleted successfully 
/go/src/sigs.k8s.io/kubefed/test/common/crudtester.go:477

[Fail] Federated "replicasets.apps" [It] should be created, read, updated and deleted successfully 
/go/src/sigs.k8s.io/kubefed/test/common/crudtester.go:519
```
It seems not releated to my pr. @hectorj2f @xunpan @irfanurrehman

**Comment by irfanurrehman** [comment]
@jimmidyson can this be related to the recent update to github actions? I did rerun this 3 times and each time, one of the first or second executed test failed with timeout (we set `single-call-timeout` of 1m for e2e test). Curiously all the later tests passed.

@shiyan2016 can you please push a "separate temperory commit" on this PR to update the [timeout](https://github.com/kubernetes-sigs/kubefed/blob/master/scripts/pre-commit.sh#L34) to a higher value, say 2m. This is only to determine if its just that the tests are executing slower then expected in the new environment, or its something else. Thanks in advance.

/hold

**Comment by shiyan2016** [comment]
> single-call-timeout

@irfanurrehman  Done for single-call-timeout to 2m. Please retriggle test. Thanks.
.

**Pull request merged**